### PR TITLE
v1.4.10: Fix resource leak in HiveSupport

### PR DESCRIPTION
**Pull request by jdcba**
Fixes resource leak in HiveSupport (#70) by shutting down Derby after each unit test. 
Adds HiveInMemoryMetastore option for HiveSupport to minimise I/O resources.

### CODE REVIEW

**Comment by jdcba** [comment]
Hi, could [this](https://github.com/CommBank/thermometer/pull/71/commits/ff33423b2a77ff164b0de8bfdb0abfce79a5b1c2) please be reviewed with high priority? I lost some days to the Travis CI issue already. I need to propagate the bugfix up through maestro, util libs, etc, as our ETL template is failing to build due to resource exhaustion and this is blocking a project.

Due to the non-functional nature of the bug for large test suites (Linux: too many open files), I don’t have a unit test to check for leaking file descriptors in the JVM. However, this patch has been tested with over 100 tests in one of our in house repos.

This PR also contains 3 commits that were required to overcome a Travis CI config issue. These are not related to the resource leak, but were necessary to get this PR to build. They also prove the nature of an enduring solution for the bugs in `.travis.yml` and upstream in ci2.

File: .travis.yml [hunk 1]
```diff
@@ -9,19 +9,19 @@ cache:
...
 after_script:
 - rm -rf ci
 env:
   global:
   - secure: tWL8CdYcRh3UUvDmiWZi+WrZBKUk9QajGZNgt4bu+NOvE8kHiibMioDZGfAtPQZaG+A0CXhOMLgcGsbmJFey4l5C1G4lajzh8Tz1ErkYpGAVvK9Ag6MFFrj1fR9cIHZlK6/dNdaQ0flsfFCbe8sMbj+XfLPPpZ0sVSXn7zzOwnM=
   - secure: KkUu2n1WA7Mmi/PO660SejCv7UwEMNJdkzCKvmADUgXkLKQQtOJr96QPwUMiTA/9LvsshQvZhamud/iwJLsfESRP+alEy+UqHpitQuMXkuy6O/ToAzPB/XgD4mpxeuiEWu8WL+ar1wPI+bNXnRcFcy1uhp7Y+NpH+5V5zzfVfJg=
 before_install:
-- openssl aes-256-cbc -K $encrypted_0f1f0cf625a6_key -iv $encrypted_0f1f0cf625a6_iv
-  -in .ci/deploy-key.enc -out .ci/deploy-key.pem -d
+- '[[ -z "$encrypted_0f1f0cf625a6_iv" ]] || openssl aes-256-cbc -K $encrypted_0f1f0cf625a6_key -iv $encrypted_0f1f0cf625a6_iv
```
Comment (rowandavies) [inline .travis.yml]: Would checking TRAVIS_PULL_REQUEST be enough?  A PR from a forked repo seems like the only scenario to consider...

File: hive/src/main/scala/au/com/cba/omnia/thermometer/hive/TheremometerHiveSpec.scala [hunk 2]
```diff
@@ -28,4 +28,10 @@ abstract class ThermometerHiveSpec extends ThermometerSpec with HiveSupport {
     super[ThermometerSpec].before
     super[HiveSupport].before
   }
+
```
Comment (rowandavies) [inline hive/src/main/scala/au/com/cba/omnia/thermometer/hive/TheremometerHiveSpec.scala]: Rename the file `Theremometer` -> `Thermometer`.

File: .travis.yml (hunk 1 above)
Comment (jdcba) [inline .travis.yml]: There are two separate-but-related issues: forks and PRs. The previous checks are for PRs, this one is for forks. I will make a tech debt note to revisit this when ci2 has been patched upstream.

**Comment by andynygard** [comment]
Just missed the merge - in hindsight, this change looks good to me. I see you've raised an issue in the `ci2` repo to be resolved at the source, so 👍

**Pull request merged**